23_multi-assay_analyses.Rmd

```diff
@@ -0,0 +1,169 @@
+of complementary data sources in a single reproducible workflow. 
+
+Another experiment can be stored in 
+[altExp](https://microbiome.github.io/OMA/containers.html#alternative-experiments) 
+slot of SE data container or both experiments can be stored side-by-side in
+[MAE](https://microbiome.github.io/OMA/containers.html#multiassayexperiments) 
+data container. 
+
+Different experiments are first imported into SE or TreeSE data container similarly
+to the case when only one experiment is present. After that different experiments are 
+combined into the same data container. Result is one TreeSE object with alternative
+experiment in altExp slot, or MAE object with multiple experiment in its 
+experiment slot. 
+
+As an example data, we use data from following publication: Hintikka L _et al._ (2021) 
+[Xylo-oligosaccharides in prevention of hepatic steatosis and adipose tissue inflammation: 
+associating taxonomic and metabolomic patterns in fecal microbiotas with 
+biclustering](https://doi.org/10.3390/ijerph18084049).
+
+In this article, mice were fed with high-fat and low-fat diets with or without prebiotics.
+The purpose of this was to study if prebiotics would reduce the negative impacts
+of high-fat diet. 
+
+This example data can be loaded from microbiomeDataSets. The data is already in MAE
+format. It includes three different experiments: microbial abundance data, 
+metabolite concentrations, and data about different biomarkers. Help for importing
+data into SE object you can find from [here](https://microbiome.github.io/OMA/containers.html#loading-experimental-microbiome-data).
+
+```{r cross-correlation1}
+# Load the data
+mae <- microbiomeDataSets::HintikkaXOData()
+
+mae
+```
+```{r cross-correlation2}
+if(!require(stringr)){
+    install.packages("stringr")
+    library(stringr)
+}
+# Drop off those bacteria that do not include information in Phylum or lower levels
+mae[[1]] <- mae[[1]][!is.na(rowData(mae[[1]])$Phylum), ]
+# Clean taxonomy data, so that names do not include additional characters
+rowData(mae[[1]]) <- DataFrame(apply(rowData(mae[[1]]), 2, 
+                                     str_remove, pattern = "._[0-9]__"))
+# Microbiome data
+mae[[1]]
+```
+
+```{r cross-correlation3}
+# Metabolite data
+mae[[2]]
+```
+
+```{r cross-correlation4}
+# Biomarker data
+mae[[3]]
+```
+
+Next we can do the cross-correlation analysis. 
+Here we analyse if individual bacteria genera correlates
+with concentrations of individual metabolites. This helps as to answer the question: 
+"If this bacteria is present, is this metabolite's concentration then low or high"?
+
+```{r cross-correlation5}
+# Agglomerate microbiome data at Genus level
+mae[[1]] <- agglomerateByRank(mae[[1]], rank = "Genus")
+# Does log10 transform for microbiome data
+mae[[1]] <- transformSamples(mae[[1]], method = "log10", pseudocount = 1)
+
+# Cross correlates data sets
+correlation_table <- testForExperimentCrossCorrelation(mae, 
```

Comment:
Just started to think if we could/should have a shorter name like `testExperimentCrossCorrelation`.. is the "For" important?

Reply:
shorter name might be better indeed, I have some other improvement ideas also so I can make a PR where we can discuss about these